qek/kernel/kernel.py

```diff
@@ -49,59 +52,113 @@ class from qek_os.data_io.dataset. The size_max parameter controls the maximum
         js = (
             jensenshannon(p=dist_graph_1, q=dist_graph_2) ** 2
         )  # Because the divergence is the square root of the distance
-        return float(np.exp(-self.mu * js))
+        return float(np.exp(-self.params["mu"] * js))
 
-    def create_train_kernel_matrix(self, train_dataset: Sequence[ProcessedData]) -> np.ndarray:
-        """Compute a kernel matrix for a given training dataset.
+    def fit(self, train_dataset: Sequence[ProcessedData]) -> None:
+        """Fit the kernel to the training dataset by storing the dataset.
 
-        This method computes a symmetric N x N kernel matrix from the Jensen-Shannon
-        divergences between all pairs of graphs in the input dataset. The resulting matrix
-        can be used as a similarity metric for machine learning algorithms.
         Args:
-            train_dataset (Sequence[ProcessedData]): A list of ProcessedData objects to compute
-            the kernel matrix from.
+            train_dataset (Sequence[ProcessedData]): The training dataset.
+        """
+        self.train_dataset = train_dataset
+        self.kernel_matrix = self._create_kernel_matrix(self.train_dataset)
```

Comment:
So, out of curiosity, why don't we make this part of `__init__`? Is this a standard practice in machine learning?